### ISSUE TITLE
PUT with multipart/form-data and array notation

### ISSUE THREAD

**Issue by mbn18** [zfcampus/zf-content-negotiation#59]
I noticed that the request is not parsed properly, when the method is PUT and the content is multipart/form-data. In that cast the arrays are squashed leaving only one value.

for example I pass in the form 2 `param[]` values. when I use `POST` I get a nice array of `$param`.
With `PUT` I get `$param[]` (the var name include []) and the only one of the values.

Here is an example of a request to an ApiGility Rest service. I just dumped the `$data` in the `WhateverResource.php`:

**notice the test param**

```
$ http --form -f PUT http://127.0.0.1/api/post/ZDqojJqaYumqxJKzXUvB7Y \
author='author' content='content' title='title' test[]='ffff' test[]='zzzz' \
file@~/Pictures/Screenshot\ from\ 2015-09-13\ 17-12-02.png

HTTP/1.1 200 OK
Connection: keep-alive
Content-Encoding: gzip
Content-Type: text/html; charset=UTF-8
Date: Tue, 09 Feb 2016 19:42:48 GMT
Server: nginx/1.9.7
Transfer-Encoding: chunked
X-Powered-By: PHP/7.0.2

object(stdClass)#531 (5) {
  ["author"]=>
  string(6) "author"
  ["content"]=>
  string(7) "content"
  ["title"]=>
  string(5) "title"
  ["test[]"]=>
  string(4) "zzzz"
  ["file"]=>
  array(5) {
    ["error"]=>
    int(0)
    ["name"]=>
    string(39) "Screenshot from 2015-09-13 17-12-02.png"
    ["type"]=>
    string(9) "image/png"
    ["tmp_name"]=>
    string(14) "/tmp/zfcmuLINJ"
    ["size"]=>
    int(8865)
  }
}
```

When the method is POST all work as expected.

**Comment by mbn18** [comment]
Here is the output of the `getContent()` function from the package `zend-http`, file `Reques.php`

```
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="title"

This is the post title
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="author"

Nimrod
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="intro"

The Intro text
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="content"

This is the WYSIWYG content <<%%image.jpg%%>> and then <<%%sparrow.jpg%%>>
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="image_title[]"

title of first image
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="image_title[]"

title of second image
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="location[]"

guides
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="location[]"

articles
------WebKitFormBoundarymy9iLjblANRI4TTg
Content-Disposition: form-data; name="created_at"

2015-03-19
------WebKitFormBoundarymy9iLjblANRI4TTg--
```

**Comment by Michal-sk** [comment]
try if this helps: https://github.com/Thunnissen/zf-content-negotiation

**Comment by solcre** [comment]
Hi @mbn18, we encountered the same bug. 

[Here](https://github.com/solcre/zf-content-negotiation/blob/hotfix/multipart-content-parser-put-array/src/MultipartContentParser.php) is how we fixed it. Instead of calling set method on $data, we append it to a string that we later parse relying on parse_str function which handles this perfectly. We didn't want to create our own parser.

We sent a pull request that we hope @weierophinney likes to merge since it's kinda simple.
